spark/dl/src/test/scala/com/intel/analytics/bigdl/optim/DistriOptimizerSpec.scala

```diff
@@ -136,6 +133,52 @@ class DistriOptimizerSpec extends FlatSpec with Matchers with BeforeAndAfter {
     }
   }
 
+  "DistriOptimizer" should "train all minibatches per epoch" in {
+    val numSamples = 64
+    val numClasses = 3
+    val height = 32
+    val width = 32
+    val images = Array.tabulate(64) { i =>
```

Comment:
I think this Unit Test won't cause the bug described in #1417 . You should create a `val miniBatches: Array[MiniBatch]` here, and create DataSet using `val dataset = DataSet.rdd(miniBatches, sc)`.